### DISCUSSION TITLE
Impossible to stop traffic

### DISCUSSION THREAD

**Discussion by CBGTI** [tnodir/fort#29] [Q&A] [answered]
Even If I enable "Stop traffic" and "Stop internet traffic", I always can go to the web with Firefox or Chrome.
I'm under Fort 3.5.6 and Windows 11 Pro

**Comment by tnodir** [comment]
What are your options for 1. & 2. in the following screenshot?

![image](https://user-images.githubusercontent.com/303655/168477184-0b89189a-f435-4c10-a0bb-b66d079433b0.png)

(I guess, you've installed as portable, but didn't install the driver.)

**Comment by CBGTI** [comment]
I have the same setup.

![Fort Cut Web](https://user-images.githubusercontent.com/89838333/168482844-25fec31c-bdf6-46e3-a03f-d1a32cce78fc.jpg)

**Comment by tnodir** [comment]
Do you use some VPN or Proxy?
If yes, then try to turn on the "Filter Local Addresses" flag and remove the "127.0.0.0/8" line from IPv4 Addresses: Internet Adresses: Exclude.

(Now my guess is based on [filtering-steps](https://github.com/tnodir/fort/wiki/FAQ#filtering-steps)).

**Comment by tnodir** [comment]
Also do you use an [IPv6 protocol](https://github.com/tnodir/fort/wiki/FAQ#state-of-ipv6)?

**Comment by CBGTI** [comment]
No proxy and no VPN.
I don't know if it's important, but I use the DNS of Google (ivp4 and ivp6).

**Comment by CBGTI** [comment]
How to be sure that my connection really use ivp4 or ivp6 protocol ?

**Comment by tnodir** [comment]
> How to be sure that my connection really use ivp4 or ivp6 protocol ?

You can use [Process Hacker 3](https://processhacker.sourceforge.io/nightly.php)'s Firewall tab to monitor WFP events: look at "Action"="ALLOW" event for a program and check its "Protocol" column. Also it'll show which provider allowed it by "Rule" column.

**Comment by CBGTI** [comment]
I have TCP and UDP for ALLOW lines.
I also have lots of red lines (DROP) with, sometimes ICMPV6 protocol. 
But I don't know wht it can be!  :-)
![TCP et UDP (ALLOW)](https://user-images.githubusercontent.com/89838333/168631299-8a3c676b-1006-4f08-a2dc-1fb9c0451454.jpg)
![ICMPV6 (Drop)](https://user-images.githubusercontent.com/89838333/168631303-478cb324-3094-482d-a18c-19899b2babd9.jpg)

**Comment by tnodir** [comment]
As you can see from 1-st screenshot, Chrome and Firefox are connected to IPv6 Remote Addresses.
For ex., Chrome connected to "2a01:e0a:...".

**Comment by CBGTI** [comment]
Right. Is it a problem?
And I can't block program with IPv6 protocol ?

**Comment by tnodir** [comment]
Yes, that's a problem until I add IPv6 addresses handling.
Sorry.

**Comment by tnodir** [comment]
Depends on #24

**Comment by CBGTI** [comment]
Ok... Thanks for your hard work and explanation.  ;-)

**Comment by tnodir** [comment]
@CBGTI Please check the new [v3.5.7](https://github.com/tnodir/fort/releases/tag/v3.5.7).

IPv6 support was added in v3.5.7 for filtering only.
Speed limiting is not supported.

**Reply by CBGTI** [reply]
Thanks for the support of IPv6.
But in my case, it don't stop traffic.

**Reply by tnodir** [reply]
Can you please take a screenshot of Process Hacker’s Firewall tab with allowed events for IPv6?

**Reply by tnodir** [reply]
By other look at your screenshots, it's strange that "Rule" for IPv4 connections are "Default outbound" too.
But the "Rule" should be "FortFilterConnect4" (and "FortFilterConnect6" for IPv6).

Do you have also some other installed firewall?
Does the Fort Firewall show traffic graph?

**Reply by CBGTI** [reply]
I don't have any other Firewall than yours.
Do you think that it's a problem/hole in the security of my computer?

For the graph, It don't work.

![Fort Firewall 1](https://user-images.githubusercontent.com/89838333/170485107-24762c6d-7df4-436d-8027-4623652094e9.jpg)
![Fort Firewall 2](https://user-images.githubusercontent.com/89838333/170485115-086359ec-68e3-4741-ac31-754c9e28cca4.jpg)
![Fort Firewall 3](https://user-images.githubusercontent.com/89838333/170485120-ff899a84-d0a0-4ead-95c4-33f4c8d674cc.jpg)
![Fort Firewall 4](https://user-images.githubusercontent.com/89838333/170485127-61c7b3fc-c0d8-4db6-a75f-bafd3f0e010a.jpg)
![Fort Firewall 5](https://user-images.githubusercontent.com/89838333/170485131-58f6ce8b-e7bb-4f3c-b955-0d351ce7284f.jpg)
![Fort Firewall Graph](https://user-images.githubusercontent.com/89838333/170485138-79d6d636-87d5-4e11-b6d5-6977bf5bde4a.jpg)
![Fort Firewall IP](https://user-images.githubusercontent.com/89838333/170485719-fc7f1829-515f-404c-80bb-3e5a654f2a07.jpg)
![Fort Firewall Pgm](https://user-images.githubusercontent.com/89838333/170485721-45fd7aaa-44a3-48d6-a845-9b30f6e08677.jpg)

**Reply by tnodir** [reply]
Very strange. There are some FortFilterConnect6 rules for local addresses, but no FortFilterConnect4 rules.

Which Windows version? 32- or 64-bit? Maybe ARM?

There is a Bandwidth indicator near of tray.
Which program is it?

What other programs do you have running?

(I've to reproduce this bug in VM.)

**Reply by tnodir** [reply]
Also what does the Statistics: Blocked Connections window contain?

Make sure that Options: Statistics: "Collect blocked connections" option is turned on.

**Reply by tnodir** [reply]
> There is a Bandwidth indicator near of tray.
> Which program is it?

Ok, I've found the [TrafficMonitor](https://github.com/zhongyang219/TrafficMonitor/).
It does **not** affect Fort's filtering.

**Reply by tnodir** [reply]
Also please check the Logs folder for errors.

**Comment by tnodir** [comment]
@CBGTI Please try the new [v3.5.8](https://github.com/tnodir/fort/releases/tag/v3.5.8) and check the `Event Viewer`: `Windows Logs`: `System` for events from Source="_**fortfw**_".

**Reply by CBGTI** [reply]
Hi. 
I instelled the v3.5.8.
I checked the Event Viewer.
Is it enough or do you need another view?

![Fort 1](https://user-images.githubusercontent.com/89838333/172383637-df53d1aa-b248-437d-a7ed-5abbb94acc54.jpg)
![Fort 2](https://user-images.githubusercontent.com/89838333/172383646-6d3d0edd-e872-4a75-a4e4-bad491e8f1fd.jpg)

**Reply by tnodir** [reply]
Hi, thanks for checking it.
There should be Error event (not Information) with "fortfw" source.

Also please check the `C:\ProgramData\Fort Firewall\logs` folder for log files.

**Comment by CBGTI** [comment]
I don't have any error with "Fortfw" in the Win11 Events Viewer.

![Fort 3](https://user-images.githubusercontent.com/89838333/172401017-784b7635-905f-439b-8269-4783f9f219c6.jpg)
![Fort 4](https://user-images.githubusercontent.com/89838333/172401026-49198636-e0c7-43ff-97c8-213037f80377.jpg)

But I have somethting in the Logs of Fort :

"2022-06-07 11:39:47.725 I Fort Firewall v3.5.8
2022-06-07 11:39:47.725 W rpc: "Configuration Error" : "Bad Exclude IP address: Error at line 1: Bad mask"

I have this line in all the logs stored in this folder (3.5.6 and over)

**Reply by tnodir** [reply]
```
"2022-06-07 11:39:47.725 I Fort Firewall v3.5.8
2022-06-07 11:39:47.725 W rpc: "Configuration Error" : "Bad Exclude IP address: Error at line 1: Bad mask"

I have this line in all the logs stored in this folder (3.5.6 and over)
```
Ok, we found it finally!
But it's still strange error. Maybe it's due OS Locale (French).
I've to try to change system language to French...

**Comment by CBGTI** [comment]
If I can help.  ;-)
Do you think that Fort is filtering something on my computer  actually ?

I already posted these screenhoots, but just to be sure :  no link with my settings ? 

![Fort5](https://user-images.githubusercontent.com/89838333/172414594-c409efbf-0b01-4932-8537-d40369f5bdcb.jpg)
![Fort6](https://user-images.githubusercontent.com/89838333/172414596-0b4bedde-b308-46d5-9cf4-c4fa467d2dc7.jpg)

**Reply by tnodir** [reply]
> Do you think that Fort is filtering something on my computer actually ?

No, it's not filtering.
Because it can't build settings for driver: failing on IPv4 address "0.0.0.0/32" parsing.

I've added French language pack to Windows 11 and changed everything to French, but still can't reproduce the bug...

**Comment by tnodir** [comment]
@CBGTI I've improved error messages for IP address parsing.
Please try the new [v3.5.9-test01](https://github.com/tnodir/fort/releases/tag/v3.5.9-test01) and show Logs for it.
Thanks.

**Comment by CBGTI** [comment, marked as answer]
I just installed it (v3.5.9 Test) and restarted my computer.

Here is my last log :
2022-06-07 20:22:50.371 I Fort Firewall v3.5.9
2022-06-07 20:22:50.371 W rpc: "Configuration Error" : "Bad Exclude IP address: Error at line 1: Bad mask. Details: [IPv4 mask='443' nbits='443' line='40.90.9.180/443']"

**Reply by tnodir** [reply]
Can you please show the `Options`: `IP Addresses`: `Allowed Internet Addresses`: `Exclude`?

It contains `40.90.9.180/443` as shown from logs.
It's an incorrect address definition, why did you enter it?

Maybe you wanted to enter port number 443 for HTTPS as `40.90.9.180:443`, but it doesn't work this way.
The `Options`: `IP Addresses` should contain only [addresses with network masks](https://en.wikipedia.org/wiki/Subnetwork), not ports.

**Reply by CBGTI** [reply]
I'm not sure that I entered this one.
All the adresses I entered was to solve conncetion problem with Forza Horizon 5 servers.
It's the official game downloaded from the Microsoft Store.

I (probably) just enter these adresses :
169.254.0.0/16 
172.16.0.0/12 
192.168.0.0/16 
239.255.255.250/32
 255.255.255.255/32

![Fort7](https://user-images.githubusercontent.com/89838333/172620995-d38da7a0-a271-4400-8abf-f72bac5928da.jpg)
![Fort8](https://user-images.githubusercontent.com/89838333/172621000-2f5f25fa-a9b4-42fe-b3f7-b2265c3832f8.jpg)

I don't know what is the difference between ":" and "/".
I just wanted to bypass Fort Firewall for these some adresses and/or programs.

**Reply by tnodir** [reply]
Please remove these 4 incorrect addresses from `Allowed Internet Addresses`: `Exclude`.

**Comment by CBGTI** [comment]
Another thing :  It could be a problem if Fort isn't installed in the default Folder?
It's on my 2nd HDD (like all my program and games).

**Reply by tnodir** [reply]
No, it's not a problem.

Just remove those 4 incorrect addresses from `Allowed Internet Addresses`: `Exclude`.

**Comment by CBGTI** [comment]
It's done.

**Reply by tnodir** [reply]
Ok, thank you for finding the bug.

I've to fix the Options' saving in error state in Service mode.

**Reply by CBGTI** [reply]
Thanks for your hard work !

**Comment by CBGTI** [comment]
Is it any way to know why I have now a symbol in front of my Fort icon near the clock?

![Fort10](https://user-images.githubusercontent.com/89838333/172861781-9b10d170-5b5e-4dab-b1ac-bb270606ae9d.JPG)



Is it any link with my log ?  -->
"2022-06-07 20:22:50.371 I Fort Firewall v3.5.9
2022-06-07 20:22:50.371 W rpc: "Configuration Error" : "Bad Exclude IP address: Error at line 1: Bad mask. Details: [IPv4 mask='443' nbits='443' line='40.90.9.180/443']"
2022-06-08 14:24:59.573 W util.net.netDownloader: NetDownloader: Error: QNetworkReply::HostNotFoundError "Host api.github.com not found"
2022-06-08 14:24:59.573 W QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once."

**Reply by tnodir** [reply]
> Is it any way to know why I have now a symbol in front of my Fort icon near the clock?

It's an Alert symbol.
You can open the `Programs` window to see new programs in Alert state.

**Comment by CBGTI** [comment]
Oups... I forget this information !  :-/

**Comment by CBGTI** [comment]
Just to confirm :  All is working now for me. 
I can (really) block web traffic for each program I want.

Thanks for the fix !   Good job!

**Reply by tnodir** [reply]
I'm glad it works for you.
